lib/linter-reek.js

```diff
@@ -19,6 +16,41 @@ const loadDeps = () => {
   }
 };
 
+const parseReekSyntaxError = (error, filePath) => {
+  const exceptionMessage = /Exception message:\s*(.*)/g.exec(error);
+  return [{
+    severity: 'error',
+    excerpt: exceptionMessage ? `linter-reek: ${exceptionMessage[1]}` : 'linter-reek: unexpected error',
+    location: {
+      file: filePath,
+      // first line of the file
+      position: [[0, 0], [0, Infinity]],
+    },
+  }];
+};
+
+const reekSmellToLinter = (smell, file, editor) => smell.lines.map(line => ({
+  url: smell.documentation_link,
+  description: () => RuleHelpers.getRuleMarkDown(smell.smell_type, smell.documentation_link),
+  excerpt: `${smell.smell_type}: ${smell.context} ${smell.message}`,
+  severity: 'warning',
+  location: {
+    file,
+    position: helpers.generateRange(editor, line - 1),
+  },
+}));
+
+const parseFromStd = (stdout) => {
+  let parsed;
+  try {
+    parsed = JSON.parse(stdout);
+  } catch (error) {
+    // continue regardless of error
+  }
+  if (typeof parsed !== 'object') { throw new Error(stdout); }
```

Comment:
```suggestion
  if (typeof parsed !== 'object') {
    throw new Error(stdout);
  }
```

Reply:
Is this expected to be an error message from `reek`? What happens if the JSON is really invalid and simply failed to parse?

Reply:
stdout is expected to be a valid JSON response, maybe this should be better

```javascript
const parseJSON = (stdout) => {
  let parsed;
  try {
    parsed = JSON.parse(stdout);
  } catch (error) {
    atom.notifications.addError('linter-reek: Unexpected error', { description: error.message });
  }
  return parsed;
};
```

Reply:
If we're expecting that to _always_ be valid JSON that is definitely better.